### PR TITLE
Add game tests for Block Reader peripheral (related to #562)

### PR DESCRIPTION
**Pull request by dogboy21**
**PLEASE READ THE [GUIDELINES](https://github.com/SirEndii/AdvancedPeripherals/blob/dev/1.20.1/CONTRIBUTING.md) BEFORE MAKING A CONTRIBUTION**

* **Please check if the PR fulfills these requirements**
- [x] The commit message are well described
- [ ] Docs have been added / updated (for features or maybe bugs which were noted). If not, please update the needed documentation [here](https://github.com/SirEndii/Advanced-Peripherals-Documentation/pulls). This is not mandatory
- [x] All changes have fully been tested

* **What kind of change does this PR introduce?** (Bug fix, feature, ...)
Feature

* **What is the current behavior?** (You can also link to an open issue here)
#562 currently does not include a test for the Block Reader peripheral.

Referring to [the comment](https://github.com/IntelligenceModding/AdvancedPeripherals/pull/562#issuecomment-2099193888) of @SirEndii:
> If someone wants to create tests, feel free to do so😅 Would speed things up
For non collaborators, you could create a PR which targets the feat/tests branch. But I am not too sure how well this goes

* **What is the new behavior (if this is a feature change)?**
This PR adds a game test for the Block reader peripheral. It tests all four functions of the peripheral with 3 different block types: simple block (Polished Andesite), stair block (to test the block state functions) and a sign block (to test the Tile Entity functions)

### CODE REVIEW

File: src/testMod/resources/data/advancedperipheralstest/computer/tests/peripheraltest.blockreader.lua [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+test.eq("minecraft:oak_sign", signReader.getBlockName(), "Block Name should be polished_andesite")
+test.neq(nil, signReader.getBlockData(), "Block Data should not be nil")
+test.assert(signReader.isTileEntity(), "Block should be a TileEntity")
+test.eq(2, tablelength(signReader.getBlockStates()), "Block State should not be empty")
+
+test.eq(4, signReader.getBlockStates()["rotation"], "Sign Rotation should be 4")
+test.assert(not signReader.getBlockStates()["waterlogged"], "Sign Waterlogged should be false")
+
+test.eq("black", signReader.getBlockData()["Color"], "Sign Color should be black")
+test.eq("{\"text\":\"this\"}", signReader.getBlockData()["Text1"], "Sign Text1 should be 'this'")
```
Comment (zyxkad) [inline src/testMod/resources/data/advancedperipheralstest/computer/tests/peripheraltest.blockreader.lua:51]: These texts seems longer than what a sign usually can hold 👀

Comment (dogboy21) [reply src/testMod/resources/data/advancedperipheralstest/computer/tests/peripheraltest.blockreader.lua:51]: The sign only has "this", "is a", "test", "sign" on the four lines, but in the sign NBT these lines are stores as a JSON chat component: https://github.com/dogboy21/AdvancedPeripherals/blob/blockreader-tests/src/testMod/resources/data/advancedperipheralstest/structures/peripheraltest.blockreader.snbt#L68

Comment (zyxkad) [reply src/testMod/resources/data/advancedperipheralstest/computer/tests/peripheraltest.blockreader.lua:51]: Ah ok, I was confused

File: src/testMod/resources/data/advancedperipheralstest/computer/tests/peripheraltest.blockreader.lua [hunk 2]
```diff
@@ -0,0 +1,54 @@
+---
+--- Advanced Peripherals Block Reader tests
+--- Covers `getBlockName`, `getBlockData`, `getBlockStates`, `isTileEntity`
+---
+
+function tablelength(T)
```
Comment (zyxkad) [inline src/testMod/resources/data/advancedperipheralstest/computer/tests/peripheraltest.blockreader.lua]:
```suggestion
function tableLength(T)
```

**Review by SirEndii — APPROVED** [review]
Looks good to me. Well written, a lot of variations tested, well done

**Pull request merged**